src/librustc/ich/fingerprint.rs

```diff
@@ -46,6 +49,21 @@ impl Fingerprint {
         format!("{:x}{:x}", self.0, self.1)
     }
 
+    pub fn encode_opaque(&self, encoder: &mut Encoder) -> EncodeResult {
+        let bytes: [u8; 16] = unsafe { mem::transmute((self.0, self.1)) };
```

Comment:
Since we are emitting raw bytes here we can't rely on leb128 to take care integer endianess, so we have to do it ourselves. Fortunately that should be straightforward. Just write the following here:
```rust
let bytes: [u8; 16] = unsafe { mem::transmute((self.0.to_le(), self.1.to_le())) };
```
and...